components/support/utils/src/main/java/mozilla/components/support/utils/WebURLFinder.kt

```diff
@@ -26,16 +26,21 @@ class WebURLFinder {
         if (string == null) {
```

Comment:
I need more time to look at this PR but I would advocate for removing `WebURLFinder` entirely – it's a complex class and cutting it apart is challenging (can we be sure we eliminated all the performance problems?) – and replacing all of its call sites with something else.